src/lib/support/CHIPCounter.h

```diff
@@ -45,20 +45,27 @@ class Counter
     virtual ~Counter() {}
 
     /**
-     *  @brief
-     *  Advance the value of the counter.
+     *  @brief Advance the value of the counter.
      *
      *  @return A CHIP error code if anything failed, CHIP_NO_ERROR otherwise.
      */
     virtual CHIP_ERROR Advance() = 0;
 
     /**
-     *  @brief
-     *  Get the current value of the counter.
+     *  @brief Get the current value of the counter.
      *
      *  @return The current value of the counter.
      */
     virtual T GetValue() = 0;
+
+    /**
+     * @brief Set counter to a new value
+     *
+     * @param value new counter value
+     *
+     * @return CHIP_ERROR A CHIP error code if anything failed, CHIP_NO_ERROR otherwise.
+     */
+    virtual CHIP_ERROR SetValue(T value) = 0;
```

Comment:
If this is meant for tests only, should it only be supported when compiling for unit tests, perhaps?  Either entirely ifdefed out, or just implemented here to return CHIP_ERROR_NOT_IMPLEMENTED and the subclass impls are conditional?

It's rather worrying to have something that can violate a counter's invariants like this, in general.

If the goal is to act as if the counter has advanced a bunch, should we have an "advance by N" method instead?

Reply:
Ah, good point about test event trigger.